core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -4540,8 +4540,11 @@ public RpcContext createRpcContext() throws UnavailableException {
 
   private LockingScheme createLockingScheme(AlluxioURI path, FileSystemMasterCommonPOptions options,
       LockPattern desiredLockMode) {
-    boolean shouldSync =
-        mUfsSyncPathCache.shouldSyncPath(path.getPath(), options.getSyncIntervalMs());
+    // If client options didn't specify the interval, fallback to whatever the server has
+    // configured to prevent unnecessary syncing due to the default value being 0
+    long syncInterval = options.hasSyncIntervalMs() ? options.getSyncIntervalMs() :
```

Comment:
so the default proto will return 0 for `get` but `hasSyncIntervalMs` will return `false`?

Reply:
Yes, that is correct. If the client specifically set `0` for the interval, then `hasSyncIntervalMs` will return `true`